### PR TITLE
Use webpack-config with wp-scritps

### PR DESCRIPTION
**Pull request by nosolosw**
- Testbed for https://github.com/WordPress/gutenberg/pull/13814
- See also https://github.com/nosolosw/understanding-gutenberg/pull/5

This uses webpack-config package with `@wordpress/scripts`. It requires that this plugin:

- Install necessary packages: `npm install --save-dev --save-exact <path-to-local-scripts-package>`
- Entry should be at `src/index.js`. Output will be at `build/index.js`

Note that current `webpack-config` setup doesn't support JSX.

### CODE REVIEW

**Comment by gziolo** [comment]
> Entry should be at `src/index.js`. Output will be at `dist/main.js`

I added logic which changes the output to `build/index.js` to follow the pattern used inside Gutenberg.

File: webpack-config/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const config = require( '@wordpress/scripts/config/webpack.config.js' );
```
Comment (gziolo) [inline webpack-config/webpack.config.js]: Next step, get rid of this file altogether :)

File: webpack-config/src/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { createElement } from '@wordpress/element';
+import { registerBlockType } from '@wordpress/blocks';
+
+registerBlockType( 'blockexample-plugin/block', {
+    title: 'Love',
+    icon: 'heart',
+    category: 'common',
+    edit: function() {
+        return ( createElement(
```
Comment (gziolo) [inline webpack-config/src/index.js]: Step 2, start using JSX in here :)

File: webpack-config/webpack.config.js (hunk 1 above)
Comment (nosolosw) [reply webpack-config/webpack.config.js]: yay! https://github.com/WordPress/gutenberg/pull/13877

**Pull request merged**